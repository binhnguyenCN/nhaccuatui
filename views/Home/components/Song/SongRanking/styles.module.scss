.song-ranking {
  height: 700px;
  background-color: brown;
}
