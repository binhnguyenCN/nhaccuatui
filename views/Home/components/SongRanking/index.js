import React from 'react';

import styles from './styles.module.scss';

function SongRanking() {
  return (
    <div className={styles['song-ranking']}>
      <h1>SONG RANKING</h1>
    </div>
  );
}
export default SongRanking;
