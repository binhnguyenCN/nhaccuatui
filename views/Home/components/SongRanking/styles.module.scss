.song-ranking {
  grid-area: songRanking;
  background-color: brown;
}
