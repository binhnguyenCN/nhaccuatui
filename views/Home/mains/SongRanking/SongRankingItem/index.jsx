// libs
import Link from 'next/link';
import React from 'react';
// layouts
import styles from './styles.module.scss';

const SongRankingItem = ({ thumbnail, title, artists, index }) => (
  <li className={styles['song-ranking-item']} data-testid="song-ranking-item">
    <div className={styles['song-ranking-thumbnail']}>
      {index === 1 && (
        <img
          src={thumbnail}
          width={100}
          height={100}
          alt="song ranking"
          title={title}
        />
      )}
      <div className={styles['ranking-order']}>{index}</div>
    </div>
    <div className={styles['song-ranking-info']}>
      <div className={styles['song-ranking-title']}>{title}</div>
      <div className={styles['song-ranking-artists']}>
        {artists.map((artist, notFirstItem) => (
          <Link href={artist.imageUrl || '/'} key={artist.artistId}>
            {notFirstItem ? `, ${artist.name}` : artist.name}
          </Link>
        ))}
      </div>
    </div>
  </li>
);

export default SongRankingItem;
