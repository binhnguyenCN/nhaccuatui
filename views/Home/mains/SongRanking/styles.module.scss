.song-ranking-wrapper {
  grid-area: songRanking;
}
