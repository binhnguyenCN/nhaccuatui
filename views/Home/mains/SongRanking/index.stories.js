// libs
import React from 'react';
// layouts
import SongRanking from './index';
// others
import mockData from '../../../../api/mockApi';

export default {
  title: 'Pages/Home/main/SongRanking',
  component: SongRanking,
  parameters: { songs: mockData.songRanking.song },
};
export const Template = () => <SongRanking songs={mockData.songRanking.song} />;
