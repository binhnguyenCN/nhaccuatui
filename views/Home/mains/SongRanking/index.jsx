// libs
import React from 'react';

// layouts
import CategoryTitles from '../../../../data_source/CategoryTitle';
import RankingOptions from '../../../../data_source/RankingOptions';
import CategoryTitle from '../../components/CategoryTitle';
import RankingList from '../../components/RankingList';
import RankingOption from '../../components/RankingOption';
import SongRankingItem from './SongRankingItem';

// others
import styles from './styles.module.scss';

const SongRanking = ({ songs }) => (
  <div className={styles['song-ranking-wrapper']}>
    <div className={styles['song-ranking-inner']}>
      <CategoryTitle title={CategoryTitles.songRanking} />
      <RankingOption options={RankingOptions.options} />
      <RankingList>
        {songs &&
          songs.map((song, index) => (
            <SongRankingItem song={song} index={index + 1} key={song.songKey} />
          ))}
      </RankingList>
    </div>
  </div>
);
export default SongRanking;
