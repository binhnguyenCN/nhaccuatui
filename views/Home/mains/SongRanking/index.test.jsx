// libs
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
// layouts
import SongRanking from '.';
// mocks
import mockData from '@/mocks/mockApi';

describe('render song list', () => {
  it('should include 12 items', () => {
    // arrange
    const testId = 'song-ranking-item';
    const maxLength = 10;
    // act
    render(<SongRanking songs={mockData.songRanking.song} />);
    // assert
    expect(screen.getAllByTestId(testId)).toHaveLength(maxLength);
  });
  it('should renders correctly', () => {
    // act
    const tree = renderer
      .create(<SongRanking songs={mockData.songRanking.song} />)
      .toJSON();
    // assert
    expect(tree).toMatchSnapshot();
  });
});
