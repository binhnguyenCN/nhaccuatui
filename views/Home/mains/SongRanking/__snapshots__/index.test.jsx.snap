// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render song list should renders correctly 1`] = `
<div
  className="song-ranking-wrapper"
>
  <div
    className="song-ranking-inner"
  >
    <h3
      className="category-title"
    >
      <span
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <span>
          bxh bài hát
        </span>
        <span>
          <span
            aria-label="play-circle"
            className="anticon anticon-play-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="play-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
              />
            </svg>
          </span>
        </span>
      </span>
    </h3>
    <ul
      className="ranking-option"
    >
      <li
        className="ranking-item ranking-item-active"
      >
        <button
          className="ranking-btn"
          onClick={[Function]}
          type="button"
        >
          Việt Nam
        </button>
      </li>
      <li
        className="ranking-item"
      >
        <button
          className="ranking-btn"
          onClick={[Function]}
          type="button"
        >
          Âu Mỹ
        </button>
      </li>
      <li
        className="ranking-item"
      >
        <button
          className="ranking-btn"
          onClick={[Function]}
          type="button"
        >
          Hàn Quốc
        </button>
      </li>
    </ul>
    <ul
      className="ranking-list"
    >
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <img
            alt="song ranking"
            height={100}
            src="https://avatar-ex-swe.nixcdn.com/song/2022/01/26/4/e/f/e/1643184497199_300.jpg"
            title="Yêu Đương Khó Quá Thì Chạy Về Khóc Với Anh"
            width={100}
          />
          <div
            className="ranking-order"
          >
            1
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            Yêu Đương Khó Quá Thì Chạy Về Khóc Với Anh
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2019/10/31/c/0/b/8/1572506582623_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              ERIK
            </a>
          </div>
        </div>
      </li>
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <div
            className="ranking-order"
          >
            2
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            Đế Vương
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2020/09/22/5/3/5/d/1600744344048_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Đình Dũng
            </a>
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2020/08/17/0/6/7/1/1597629837858_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              , ACV
            </a>
          </div>
        </div>
      </li>
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <div
            className="ranking-order"
          >
            3
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            Hoa Tàn Tình Tan
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2022/03/08/5/c/b/f/1646713382361_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Giang Jolee
            </a>
          </div>
        </div>
      </li>
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <div
            className="ranking-order"
          >
            4
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            Chạy Về Nơi Phía Anh
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2020/05/27/6/9/5/0/1590565041302_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Khắc Việt
            </a>
          </div>
        </div>
      </li>
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <div
            className="ranking-order"
          >
            5
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            Có Em Đây
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2020/08/12/f/2/d/1/1597212025646_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Như Việt
            </a>
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              , Dunghoangpham
            </a>
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2020/08/17/0/6/7/1/1597629837858_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              , ACV
            </a>
          </div>
        </div>
      </li>
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <div
            className="ranking-order"
          >
            6
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            có hẹn với thanh xuân
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2017/12/11/d/e/b/9/1512984325191_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              MONSTAR
            </a>
          </div>
        </div>
      </li>
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <div
            className="ranking-order"
          >
            7
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            Thức Giấc
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2020/08/03/3/f/4/5/1596425146149_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Da LAB
            </a>
          </div>
        </div>
      </li>
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <div
            className="ranking-order"
          >
            8
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            Tệ Thật, Anh Nhớ Em
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2019/11/15/2/7/c/d/1573834150811_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Thanh Hưng
            </a>
          </div>
        </div>
      </li>
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <div
            className="ranking-order"
          >
            9
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            Là Do Em Xui Thôi
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2020/10/27/e/b/6/8/1603769786589_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Khói
            </a>
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2021/01/28/0/1/6/6/1611805269799_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              , Sofia
            </a>
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2019/11/11/4/4/d/7/1573460448991_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              , Châu Đăng Khoa
            </a>
          </div>
        </div>
      </li>
      <li
        className="song-ranking-item"
        data-testid="song-ranking-item"
      >
        <div
          className="song-ranking-thumbnail"
        >
          <div
            className="ranking-order"
          >
            10
          </div>
        </div>
        <div
          className="song-ranking-info"
        >
          <div
            className="song-ranking-title"
          >
            Cưới Thôi
          </div>
          <div
            className="song-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2019/10/29/a/a/d/4/1572318457703_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Masew
            </a>
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2021/08/30/8/3/f/f/1630313898499_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              , Masiu
            </a>
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2020/05/27/6/9/5/0/1590562104987_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              , B Ray
            </a>
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              , TAP
            </a>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
